js_modules/dagit/packages/core/src/ui/Filters/FilterDropdown.tsx

```diff
@@ -23,7 +31,21 @@ interface FilterDropdownProps {
   setPortaledElements: React.Dispatch<React.SetStateAction<JSX.Element[]>>;
 }
 
+const focusedIndexAtom = atomFamily<number, string>({
+  key: 'FilterDropdown:focusedIndex',
+  default: -1,
+});
+const isFocusedSelector = selectorFamily<boolean, {key: string; index: number}>({
+  key: 'FilterDropdown:isFocused',
+  get: ({key, index}) => ({get}) => get(focusedIndexAtom(key)) === index,
+});
+
 export const FilterDropdown = ({filters, setIsOpen, setPortaledElements}: FilterDropdownProps) => {
+  const [menuKey, _] = React.useState(() => uuidv4());
+  const setFocusedItemIndex = useSetRecoilState(focusedIndexAtom(menuKey));
+  const isSearchInputFocused = useRecoilValue(
+    isFocusedSelector(React.useMemo(() => ({key: menuKey, index: -1}), [menuKey])),
+  );
```

Comment:
Is recoil necessary to manage this state? It seems fairly straightforward, and it doesn't look like any props need to be passed far down the tree.

Also see comment re: highlighting menu items while retaining focus on the input.

Reply:
Yes Recoil allowed me to remove `focusedIndex` from the `useMemo` that re-renders the results. This means that when we change focus we don't re-render the entire list. This makes keyboard navigation a lot smoother.

Reply:
Why is it so slow to rerender the results? It doesn't seem like it should be too expensive just rendering a list of keyed items, but I may be missing something.

Reply:
> Why is it so slow to rerender the results? 

It's extra render code that is running which makes it slower. Since this is supposed to be a "core" component I think it's important to optimized renders. Especially since I'm planning on making some of these filters async down the line and I wouldn't want an async request to be fired more than necessary

Reply:
Just a drive-by, but I think the solution here might be to use a virtualized list for the dropdown so that "re-render the entire list" is just 6 items? Otherwise to dish's point it still takes a while for the dropdown to appear because the initial render is still large?

Reply:
A Virtualized list will certainly help, but it's still an unnecessary extra render that can be skipped